AnalysisNtuple/Selector.cpp

```diff
@@ -323,25 +373,39 @@ void Selector::filter_electrons(){
         bool passDz = ((absEta < 1.479 && abs(tree->eleDz_[eleInd]) < 0.1) ||
                        (absEta > 1.479 && abs(tree->eleDz_[eleInd]) < 0.2));
         
-        
 
         double EleSmear = 1.;
 
-        /////////NEEDS TO BE REIMPLEMENTED
-
-        // if (!tree->isData_ && elesmearLevel==1) {EleSmear = generator->Gaus(1,(tree->eleResol_rho_up_[eleInd]+tree->eleResol_rho_dn_[eleInd])/2.);}
-        // if (!tree->isData_ && elesmearLevel==0) {EleSmear = generator->Gaus(1,tree->eleResol_rho_dn_[eleInd]);}
-        // if (!tree->isData_ && elesmearLevel==2) {EleSmear = generator->Gaus(1,tree->eleResol_rho_up_[eleInd]);}
-        // if (pt<10.){
-        //   smearEle= false;
-        // }
-        // if (smearEle){
-        //   pt = pt*EleSmear;
-        //   en = EleSmear*en;
-        // }
-        // tree->elePt_[eleInd] = pt;
-        // tree->eleEn_[eleInd]= en;   
-        // double EleScale = 1.;
+       // if (tree->event_==printEvent){ cout  << "======> pt before " << pt << endl;}
+        if (!tree->isData_ && (elesmearLevel==0 || elesmearLevel==2)) {
+        	if (pt <= 15) {pt = 16.0;}
+        	if (absEta >= 2.4) {absEta = 2.39;}
+        	EleSmear = getElePhoSmearScaleSF(pt,absEta); 
+			// EleSmear = 1.5; getEleSmearSF
+			pt = pt*EleSmear; 
+       		// if (tree->event_==printEvent){ cout  << "======> smear value " << EleSmear << endl;}
+
+
+        } 
+   
+
+        // if (tree->event_==printEvent){ cout  << "=======> pt after " << pt << endl;}
+
+        double EleScale = 1.;
+
+        if (!tree->isData_ && (elescaleLevel==0 || elescaleLevel==2)) {
+        	if (pt <= 15) {pt = 16.0;}
+        	if (absEta >= 2.4) {absEta = 2.39;}
+        	EleScale = getElePhoSmearScaleSF(pt,absEta); 
+			// EleSmear = 1.5; getEleSmearSF
+			pt = pt*EleScale; 
+       		// if (tree->event_==printEvent){ cout  << "======> smear value " << EleSmear << endl;}
+
+
```

Comment:
Same comment as line 189, also remove unneccesary commented out lines and fix tabbing on line 401.